modules/market/src/main/java/com/opengamma/strata/market/param/TenorTenorParameterMetadata.java

```diff
@@ -0,0 +1,492 @@
+/*
+ * Copyright (C) 2019 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.market.param;
+
+import java.io.Serializable;
+import java.util.Map;
+import java.util.NoSuchElementException;
+
+import org.joda.beans.Bean;
+import org.joda.beans.BeanBuilder;
+import org.joda.beans.ImmutableBean;
+import org.joda.beans.JodaBeanUtils;
+import org.joda.beans.MetaBean;
+import org.joda.beans.MetaProperty;
+import org.joda.beans.gen.BeanDefinition;
+import org.joda.beans.gen.ImmutablePreBuild;
+import org.joda.beans.gen.PropertyDefinition;
+import org.joda.beans.impl.direct.DirectMetaBean;
+import org.joda.beans.impl.direct.DirectMetaProperty;
+import org.joda.beans.impl.direct.DirectMetaPropertyMap;
+import org.joda.beans.impl.direct.DirectPrivateBeanBuilder;
+
+import com.opengamma.strata.basics.date.Tenor;
+import com.opengamma.strata.collect.tuple.Pair;
+
+/**
+ * Parameter metadata based on an expiry tenor, an underlying tenor and their respective year fractions.
+ */
+@BeanDefinition(builderScope = "private")
+public final class TenorTenorParameterMetadata
+    implements ParameterMetadata, ImmutableBean, Serializable {
```

Comment:
Maybe implement TenoredParameterMetadata? Although perhaps @jodastephen can give more insight into that